### ISSUE TITLE
[YSQL] ERROR:  index row requires ... bytes, maximum size is 8191 with

### ISSUE THREAD

**Issue by FranckPachot** [yugabyte/yugabyte-db#13108]
Jira Link: [DB-2816](https://yugabyte.atlassian.net/browse/DB-2816)
### Description

There's a limitation of 8191 row size from Index Scan which probably comes from PostgreSQL block size. It would be nice to release this limitation especially with covering indexes including a large column.
Example:
```
drop table if exists franck;
create extension if not exists orafce;
create table franck (
 n bigint
 ,c1 text default dbms_random.string('a',10000)::text
 ,c2 text default dbms_random.string('a',100000)::text
)
;
insert into franck(n) select generate_series(1,100);
create index franck_n_c1 on franck(n asc) include (c1);
-- check plan: Index Only Scan
explain (costs off, verbose)
select n,c1 from franck where n<=10;
-- execute: error
explain (costs off, summary off, analyze, verbose)
select n,c1 from franck where n<=10;
```
The create index works, but the Index Scan fails with:
```
yugabyte=# create index franck_n_c1 on franck(n asc) include (c1);
CREATE INDEX
yugabyte=# explain (costs off, verbose)
yugabyte-# select n,c1 from franck where n<=10;
                     QUERY PLAN
----------------------------------------------------
 Index Only Scan using franck_n_c1 on public.franck
   Output: n, c1
   Index Cond: (franck.n <= 10)
(3 rows)

yugabyte=# explain (costs off, summary off, analyze, verbose)
yugabyte-# select n,c1 from franck where n<=10;
ERROR:  index row requires 10040 bytes, maximum size is 8191
```

**Comment by ddorian** [comment]
Assigned after talking with Mihnea.

**Comment by lnguyen-yugabyte** [comment]
The fix applied for #12845 also works for this case. Previewing the output:

```
yugabyte=# explain (costs off, verbose)
yugabyte-# select n,c1 from franck where n<=10;
                     QUERY PLAN
----------------------------------------------------
 Index Only Scan using franck_n_c1 on public.franck
   Output: n, c1
   Index Cond: (franck.n <= 10)
(3 rows)

yugabyte=# explain (costs off, summary off, analyze, verbose)
yugabyte-# select n,c1 from franck where n<=10;
                                          QUERY PLAN
-----------------------------------------------------------------------------------------------
 Index Only Scan using franck_n_c1 on public.franck (actual time=2.054..2.160 rows=10 loops=1)
   Output: n, c1
   Index Cond: (franck.n <= 10)
   Heap Fetches: 0
(4 rows)
```

**Comment by lnguyen-yugabyte** [comment]
Resolved with commit https://github.com/yugabyte/yugabyte-db/commit/2860a1b634b786d418d3ae9744e74b03b139a903
